xchange-ftx/src/main/java/org/knowm/xchange/ftx/service/FtxLendingService.java

```diff
@@ -0,0 +1,277 @@
+  }
+
+  public List<FtxLendDataDto> stopLending(String subaccount, String... coins) {
+    return stopLending(subaccount, Arrays.asList(coins));
+  }
+
+  public FtxLendDataDto lendAll(String subaccount, String coin, double rate) {
+    Objects.requireNonNull(coin);
+    if (StringUtils.isNotBlank(coin) || StringUtils.isNotEmpty(coin))
+      throw new FtxLendingServiceException("Coin are blank or empty");
+    if (rate < 0)
+      throw new FtxLendingServiceException("Rate must to be >= 0, subaccount: " + subaccount + ", coin: " + coin);
+
+    try {
+      FtxLendingInfoDto info = info(subaccount, coin);
+
+      double sizeToLend = BigDecimalUtils.ftxRoundingLending(new BigDecimal(info.getLendable())).doubleValue();
+      if (sizeToLend == info.getOffered())
+        return new FtxLendDataDto(coin, info.getLocked(), sizeToLend, sizeToLend, rate);
+
+      ftx.submitLendingOffer(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount,
+          new FtxSubmitLendingOfferParams(coin, sizeToLend, rate)
+      );
+      return new FtxLendDataDto(coin, info.getLocked(), info.getOffered(), sizeToLend, rate);
+    } catch (IOException e) {
+      throw new FtxLendingServiceException(e.getMessage());
+    }
+  }
+
+  public List<FtxLendDataDto> lendAll(String subaccount, Map<String, Double> rateByCoins) {
+    List<FtxLendDataDto> list = new ArrayList<>();
+    rateByCoins.forEach((coin, rate) -> {
+
+
+      List<FtxLendingInfoDto> infos = infos(subaccount);
+      Optional<FtxLendingInfoDto> op = infos.stream()
+          .filter(lendingInfo -> lendingInfo.getCoin().equalsIgnoreCase(coin))
+          .findFirst();
+
+      if (!op.isPresent()) throw new FtxLendingServiceException("Cant lend all, infos don't exist for coin: " + coin);
+
+      double sizeToLend = BigDecimalUtils.ftxRoundingLending(new BigDecimal(op.get().getLendable())).doubleValue();
+      if (sizeToLend == op.get().getOffered())
+        list.add(new FtxLendDataDto(coin, op.get().getLocked(), sizeToLend, sizeToLend, rate));
+      else {
+        try {
+          ftx.submitLendingOffer(
+              exchange.getExchangeSpecification().getApiKey(),
+              exchange.getNonceFactory().createValue(),
+              signatureCreator,
+              subaccount,
+              new FtxSubmitLendingOfferParams(coin, sizeToLend, rate)
+          );
+          list.add(new FtxLendDataDto(coin, op.get().getLocked(), op.get().getOffered(), sizeToLend, rate));
+        } catch (IOException e) {
+          throw new FtxLendingServiceException("Can't lend all for subaccount: " + subaccount + ", coin: " + coin + ", rate: " + rate);
+        }
+      }
+    });
+    return null;
+  }
+
+  public FtxLendDataDto lend(String subaccount, String coin, double size, double rate) {
+    Objects.requireNonNull(coin);
+    if (StringUtils.isNotBlank(coin) || StringUtils.isNotEmpty(coin))
```

Comment:
ib you are checking isnotblank you dont need objects. check or isnotempty check

Reply:
I change it